NPE on SDK setup

```
Exception while executing the process: Cannot run program "C:\go_appengine\goroot\..\goapp" (in directory "C:\go_appengine\goroot\.."): CreateProcess error=2, The system cannot find the file specified
com.intellij.execution.process.ProcessNotCreatedException: Cannot run program "C:\go_appengine\goroot\..\goapp" (in directory "C:\go_appengine\goroot\.."): CreateProcess error=2, The system cannot find the file specified
    at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:290)
    at ro.redeul.google.go.sdk.GoSdkUtil.testGoAppEngineSdk(GoSdkUtil.java:270)
    at ro.redeul.google.go.config.sdk.GoAppEngineSdkType.isValidSdkHome(GoAppEngineSdkType.java:53)
    at ro.redeul.google.go.config.sdk.GoAppEngineSdkType$2.validateSelectedFiles(GoAppEngineSdkType.java:158)
    at com.intellij.openapi.fileChooser.ex.FileChooserDialogImpl.doOKAction(FileChooserDialogImpl.java:408)
    at com.intellij.openapi.ui.DialogWrapper$OkAction.doAction(DialogWrapper.java:1791)
    at com.intellij.openapi.ui.DialogWrapper$DialogWrapperAction.actionPerformed(DialogWrapper.java:1757)
    at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2018)
    at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2341)
    at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
    at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
    at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
    at java.awt.Component.processMouseEvent(Component.java:6505)
    at javax.swing.JComponent.processMouseEvent(JComponent.java:3320)
    at java.awt.Component.processEvent(Component.java:6270)
    at java.awt.Container.processEvent(Container.java:2229)
    at java.awt.Component.dispatchEventImpl(Component.java:4861)
    at java.awt.Container.dispatchEventImpl(Container.java:2287)
    at java.awt.Component.dispatchEvent(Component.java:4687)
    at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4832)
    at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4492)
    at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4422)
    at java.awt.Container.dispatchEventImpl(Container.java:2273)
    at java.awt.Window.dispatchEventImpl(Window.java:2719)
    at java.awt.Component.dispatchEvent(Component.java:4687)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:735)
    at java.awt.EventQueue.access$200(EventQueue.java:103)
    at java.awt.EventQueue$3.run(EventQueue.java:694)
    at java.awt.EventQueue$3.run(EventQueue.java:692)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:87)
    at java.awt.EventQueue$4.run(EventQueue.java:708)
    at java.awt.EventQueue$4.run(EventQueue.java:706)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:705)
    at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:748)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:573)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:154)
    at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:182)
    at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:221)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:219)
    at java.awt.Dialog.show(Dialog.java:1082)
    at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:779)
    at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:464)
    at com.intellij.openapi.ui.DialogWrapper.showAndGetOk(DialogWrapper.java:1569)
    at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1536)
    at com.intellij.openapi.fileChooser.ex.FileChooserDialogImpl.choose(FileChooserDialogImpl.java:150)
    at com.intellij.openapi.fileChooser.FileChooser.chooseFiles(FileChooser.java:111)
    at com.intellij.openapi.fileChooser.FileChooser.chooseFiles(FileChooser.java:89)
    at com.intellij.openapi.projectRoots.impl.SdkConfigurationUtil.selectSdkHome(SdkConfigurationUtil.java:284)
    at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel.doAdd(ProjectSdksModel.java:275)
    at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel$3.actionPerformed(ProjectSdksModel.java:257)
    at com.intellij.ui.popup.PopupFactoryImpl$ActionPopupStep$1.run(PopupFactoryImpl.java:739)
    at com.intellij.ui.popup.AbstractPopup$18.run(AbstractPopup.java:1343)
    at com.intellij.openapi.wm.impl.FocusManagerImpl.a(FocusManagerImpl.java:651)
    at com.intellij.openapi.wm.impl.FocusManagerImpl.g(FocusManagerImpl.java:632)
    at com.intellij.openapi.wm.impl.FocusManagerImpl.e(FocusManagerImpl.java:602)
    at com.intellij.openapi.wm.impl.FocusManagerImpl.access$200(FocusManagerImpl.java:60)
    at com.intellij.openapi.wm.impl.FocusManagerImpl$IdleRunnable.runEdt(FocusManagerImpl.java:108)
    at com.intellij.openapi.util.EdtRunnable$1.run(EdtRunnable.java:28)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:251)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:733)
    at java.awt.EventQueue.access$200(EventQueue.java:103)
    at java.awt.EventQueue$3.run(EventQueue.java:694)
    at java.awt.EventQueue$3.run(EventQueue.java:692)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:703)
    at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:748)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:577)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:154)
    at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:182)
    at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:221)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:219)
    at java.awt.Dialog.show(Dialog.java:1082)
    at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:779)
    at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:464)
    at com.intellij.openapi.ui.DialogWrapper.showAndGetOk(DialogWrapper.java:1569)
    at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1536)
    at com.intellij.ide.actions.ShowStructureSettingsAction.actionPerformed(ShowStructureSettingsAction.java:48)
    at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$3.performAction(IdeKeyEventDispatcher.java:614)
    at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:663)
    at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.d(IdeKeyEventDispatcher.java:513)
    at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:211)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:546)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
Caused by: java.io.IOException: Cannot run program "C:\go_appengine\goroot\..\goapp" (in directory "C:\go_appengine\goroot\.."): CreateProcess error=2, The system cannot find the file specified
    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1041)
    at com.intellij.execution.configurations.GeneralCommandLine.startProcess(GeneralCommandLine.java:300)
    at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:286)
    ... 101 more
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
    at java.lang.ProcessImpl.create(Native Method)
    at java.lang.ProcessImpl.<init>(ProcessImpl.java:385)
    at java.lang.ProcessImpl.start(ProcessImpl.java:136)
    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1022)
    ... 103 more
```

--- a/src/ro/redeul/google/go/sdk/GoSdkUtil.java
+++ b/src/ro/redeul/google/go/sdk/GoSdkUtil.java
@@ -253,14 +253,27 @@ public class GoSdkUtil {
         sdkData.GO_HOME_PATH = format("%s%sgoroot", path, File.separator);
         sdkData.GO_GOPATH_PATH = getGoPath();
 
-        GeneralCommandLine command = new GeneralCommandLine();
-        if (checkFileExists(sdkData.GO_HOME_PATH + "/bin/go")) {
-            command.setExePath(sdkData.GO_HOME_PATH + "/bin/go");
-            command.withWorkDirectory(sdkData.GO_HOME_PATH + "/bin");
-        } else {
-            command.setExePath(sdkData.GO_HOME_PATH + "/../goapp");
-            command.withWorkDirectory(sdkData.GO_HOME_PATH + "/..");
+        String execName = sdkData.GO_HOME_PATH + "/bin/go";
+        if (isHostOsWindows()) {
+            execName += ".exe";
         }
+        String homePath = sdkData.GO_HOME_PATH + "/bin";
+
+        GeneralCommandLine command = new GeneralCommandLine();
+        if (checkFileExists(execName)) {
+            execName = sdkData.GO_HOME_PATH + "/../goapp";
+            if (isHostOsWindows()) {
+                execName += ".bat";
+            }
+            homePath = sdkData.GO_HOME_PATH + "/..";
+
+            if (!checkFileExists(execName)) {
+                return null;
+            }
+        }
+
+        command.setExePath(execName);
+        command.withWorkDirectory(homePath);
         command.addParameter("env");
 
         sdkData.TARGET_ARCH = GoTargetArch._amd64;

